File: python/semantic_kernel/processes/kernel_process/kernel_process_function_target.py
# Copyright (c) Microsoft. All rights reserved.

from semantic_kernel.kernel_pydantic import KernelBaseModel
from semantic_kernel.utils.feature_stage_decorator import experimental


@experimental
class KernelProcessFunctionTarget(KernelBaseModel):
    """The target of a function call in a kernel process."""

    step_id: str
    function_name: str
    parameter_name: str | None = None
    target_event_id: str | None = None

File: nodejs/semantic-kernel/processes/kernel-process/kernel-process-function-target.ts
/**
 * The target of a function call in a kernel process.
 * @experimental
 */
export class KernelProcessFunctionTarget {
  /**
   * The step ID.
   */
  stepId: string

  /**
   * The function name.
   */
  functionName: string

  /**
   * The optional parameter name.
   */
  parameterName?: string | null = null

  /**
   * The optional target event ID.
   */
  targetEventId?: string | null = null

  constructor(params: {
    stepId: string
    functionName: string
    parameterName?: string | null
    targetEventId?: string | null
  }) {
    this.stepId = params.stepId
    this.functionName = params.functionName
    if (params.parameterName !== undefined) {
      this.parameterName = params.parameterName
    }
    if (params.targetEventId !== undefined) {
      this.targetEventId = params.targetEventId
    }
  }
}
